### ISSUE TITLE
Checkpoints: Type converstion between `CheckpointResponse` and `Checkpoint`

### ISSUE THREAD

**Issue by adlrocha** [consensus-shipyard/ipc#439]
Unfortunately, we can not de-serialize directly from the RPC response of a `Checkpoint` object that Lotus provides because the RPC API returns a map serialization, while the `Checkpoint` object defined in IPC actors use `deserialize_tuple`. We haven't currently figured out the right to fully transport from a `CheckpointReponse` to a `Checkpoint` object. Even if we simplify in the future the structure of checkpoints, we should get this working as it will be key for the following two functions and the execution of cross-net messages: 
https://github.com/consensus-shipyard/ipc-agent/blob/5b4a24a3f1685c15ba72ad0970cae2d027fdb615/src/lotus/client.rs#L283-L286

https://github.com/consensus-shipyard/ipc-agent/blob/5b4a24a3f1685c15ba72ad0970cae2d027fdb615/src/lotus/client.rs#L330-L340

TL;DR: The idea is to de-serialize into a `Checkpoint` object (through maybe a `try_from` from `CheckpointReponse` so we can remove [these lines](https://github.com/consensus-shipyard/ipc-agent/blob/5b4a24a3f1685c15ba72ad0970cae2d027fdb615/src/lotus/client.rs#L347-L350) and return the right deserialized object.
